packages/superset-ui-legacy-preset-chart-nvd3/src/transformProps.js

```diff
@@ -79,6 +78,26 @@ export default function transformProps(chartProps) {
       }))
     : rawData;
 
+  const grabD3Format = targetMetric => {
+    let foundFormatter;
+    chartProps.datasource.metrics.forEach(metric => {
+      if (metric.d3format && metric.metric_name === targetMetric) {
+        foundFormatter = metric.d3format;
+      }
+    });
+
+    return foundFormatter;
+  };
+
+  if (chartProps.formData.vizType === 'pie') {
+    numberFormat = grabD3Format(chartProps.formData.metric) || numberFormat;
+  } else if (chartProps.formData.vizType === 'dual_line') {
+    yAxisFormat = grabD3Format(chartProps.formData.metric) || yAxisFormat;
+    yAxis2Format = grabD3Format(chartProps.formData.metric2) || yAxis2Format;
+  } else if (['line', 'dist_bar', 'bar', 'area'].indexOf(chartProps.formData.vizType) > -1) {
+    yAxisFormat = grabD3Format(chartProps.formData.metrics[0]) || yAxisFormat;
```

Comment:
why do we overwrite the passed in format here but we don't in the other charts?

Reply:
I just spent a bit more time looking at this one, honestly. I think overwriting is more straightforward - I'll make the others conform, which also addresses your earlier comment about the (unnecessary, really) `metricFormat` variable.

Reply:
Now that I'm looking at this with fresh eyes, I was a little heavy handed in this approach, and I'm making a tweak. 

With this logic:
`yAxisFormat = grabD3Format(chartProps, chartProps.formData.metrics[0]) || yAxisFormat;`
... the `D3 Format` from the metric actually overrides the form data for the viz. That's not cool. I'll recommit as:
`yAxisFormat = yAxisFormat || grabD3Format(chartProps, chartProps.formData.metrics[0]);`
... then the formData (if present) will act as an override of the Metric's D3 Format setting. That setting can be cleared in the UI, allowing the Metric's format to show itself. If Neither the form nor the Metric specify a format, NVD3 takes its best guess.